lib/spack/spack/compilers/__init__.py

```diff
@@ -250,21 +251,17 @@ def get_compilers(cspec):
             if mods == 'None':
                 mods = []
 
+            os = None
             if 'operating_system' in items:
                 os = spack.architecture._operating_system_from_dict(
                     items['operating_system'], platform)
-            else:
-                os = None
 
-            alias = items['alias'] if 'alias' in items else None
+            alias = items.get('alias', None)
 
-            flags = {}
-            for f in spack.spec.FlagMap.valid_compiler_flags():
-                if f in items:
-                    flags[f] = items[f]
+            compiler_flags = items.get('flags', {})
```

Comment:
This is the most important change in the context of fixing the bug.  This code was attempting to pick up flags from the `items` dictionary when it should have been picking them up from the `items['paths']` dictionary.  Making this change directly required adding some code that only pulled the correct entries from the `items['paths']` dictionary (e.g. `cflags`, `fflags`, etc.), which was a bit ugly.  I decided to update the `compilers.yaml` structure to help clean up this section and to better organize the contents of compiler configuration (the flags really didn't belong in the paths section).